apps/charterafrica/src/components/ArticlesFilter/ArticlesFilter.js

```diff
@@ -0,0 +1,101 @@
+import { Section } from "@commons-ui/core";
+import { Box, Grid, Typography, Select, MenuItem, Chip } from "@mui/material";
+import { useRouter } from "next/router";
+import React, { useEffect, useState } from "react";
+
+import { neutral } from "@/charterafrica/colors";
+import SearchInput from "@/charterafrica/components/SearchInput";
+import useFilterQuery, {
+  ALL_TAG,
+  DEFAULT_SORTING,
+} from "@/charterafrica/components/useFilterQuery";
+
+const ArticlesFilter = React.forwardRef((props, ref) => {
+  const { tags, sorting } = props;
+  const allTags = [ALL_TAG, ...tags];
+  const [sort, setSelectedSorting] = useState(DEFAULT_SORTING);
+  const [tag, setSelectedTags] = useState(ALL_TAG);
+  const [q, setQ] = useState();
+  const router = useRouter();
+  const queryParams = useFilterQuery({ sort, q, tag });
+
+  const handleTagChange = (t) => {
+    setSelectedTags(t);
+  };
+
+  useEffect(() => {
+    router.push({
+      pathname: router.asPath.split("?")[0],
+      query: queryParams,
+    });
+
+    // We don't want to listen to router changes here since we're the ones
```

Comment:
Alternatively, you could declare pathname variable outside the react hook then add it into the dependancy array.

i.e 
```js
const pathname  = router.asPath.split("?")[0]
```
but either way this is a bad practice. It could cause recursion.